app/src/main/java/com/sjh/sunflower_sksowk156/MainActivity.kt

```diff
@@ -1,46 +1,177 @@
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.pager.HorizontalPager
+import androidx.compose.foundation.pager.PagerState
+import androidx.compose.foundation.pager.rememberPagerState
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Tab
+import androidx.compose.material3.TabRow
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.input.nestedscroll.NestedScrollConnection
+import androidx.compose.ui.input.nestedscroll.NestedScrollSource
+import androidx.compose.ui.input.nestedscroll.nestedScroll
+import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.tooling.preview.Preview
-import com.sjh.sunflower_sksowk156.ui.theme.Sunflowersksowk156Theme
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.sjh.sunflower_sksowk156.core.designsystem.theme.Sunflowersksowk156Theme
+import com.sjh.sunflower_sksowk156.feature.mygarden.MyGardenScreen
+import com.sjh.sunflower_sksowk156.feature.plantlist.PlantList
+import kotlinx.coroutines.launch
 
 class MainActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContent {
             Sunflowersksowk156Theme {
-                // A surface container using the 'background' color from the theme
-                Surface(
-                    modifier = Modifier.fillMaxSize(),
-                    color = MaterialTheme.colorScheme.background
-                ) {
-                    Greeting("Android")
-                }
+                MainScreen()
             }
         }
     }
 }
 
+@OptIn(ExperimentalFoundationApi::class)
+@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
 @Composable
-fun Greeting(name: String, modifier: Modifier = Modifier) {
-    Text(
-        text = "Hello $name!",
-        modifier = modifier
-    )
+fun MainScreen() {
+    val tabs = listOf("My garden", "Plant list")
+    val pagerState = rememberPagerState(pageCount = { tabs.size })
+    val showTopBar = remember { mutableStateOf(true) }
+
+    // NestedScrollConnection을 사용하여 스크롤에 따라 TopAppBar를 숨기거나 표시
+    val nestedScrollConnection = remember {
+        object : NestedScrollConnection {
+            override fun onPreScroll(available: Offset, source: NestedScrollSource): Offset {
+                // 스크롤 방향에 따라 TopAppBar의 보이기/숨기기 상태를 변경
+                val delta = available.y
+                showTopBar.value = delta > 0 || (delta == 0f && showTopBar.value)
```

Comment:
```suggestion
                showTopBar.value = when {
                    delta > 0 -> true
                    delta != 0f -> false
                    else -> showTopBar.value
                }
```
이렇게 하면 식의 이해가 조금 더 쉬울까요?